pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -150,3 +161,22 @@ def test_setitem_dict_preserves_dtypes(self):
                 "c": float(b),
             }
         tm.assert_frame_equal(df, expected)
+
+    def test_setitem_extension_types(self):
```

Comment:
can you parameterize on scalars here (period, interval) also add Timestamp(tz=....)

Reply:
Done